From 0113f3172c3f7119c3f9aa50614c7e53412c54e7 Mon Sep 17 00:00:00 2001
From: David Plowman <david.plowman@raspberrypi.com>
Date: Fri, 27 Oct 2023 12:14:22 +0100
Subject: [PATCH] drivers: media: imx477: Set horizontal binning when disabling
 the scaler

The horizontal scaler has been disabled but actually the sensor is not
binning horizontally, resulting in images that are stretched 2x
horizontally (missing the right half of the field of view completely).

Therefore we must additionally set the horizontal binning mode. There
is only marginal change in output quality and noise levels.

Signed-off-by: David Plowman <david.plowman@raspberrypi.com>
Fixes: f075893e9b0e ("drivers: media: imx477: Disable the scaler")
---
 drivers/media/i2c/imx477.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx477.c b/drivers/media/i2c/imx477.c
index e848f64b74bd..bab0da00e0c0 100644
--- a/drivers/media/i2c/imx477.c
+++ b/drivers/media/i2c/imx477.c
@@ -607,7 +607,7 @@ static const struct imx477_reg mode_2028x1520_regs[] = {
 	{0x0385, 0x01},
 	{0x0387, 0x01},
 	{0x0900, 0x01},
-	{0x0901, 0x12},
+	{0x0901, 0x22},
 	{0x0902, 0x02},
 	{0x3140, 0x02},
 	{0x3c00, 0x00},
@@ -708,7 +708,7 @@ static const struct imx477_reg mode_2028x1080_regs[] = {
 	{0x0385, 0x01},
 	{0x0387, 0x01},
 	{0x0900, 0x01},
-	{0x0901, 0x12},
+	{0x0901, 0x22},
 	{0x0902, 0x02},
 	{0x3140, 0x02},
 	{0x3c00, 0x00},
-- 
2.34.1

